src/utils.ts

```diff
@@ -898,97 +898,224 @@ export function deepCopy<T>(value: T): T {
+    sentinel.prev = sentinel as unknown as ListNode<T>;
+
+    this.head = sentinel;
+  }
+
+  toArray() {
+    return Array.from(this);
+  }
+
+  toString() {
+    return `head <=> ${this.toArray().join(' <=> ')} <=> head`;
+  }
+
+  *[Symbol.iterator](): Generator<T, void, void> {
+    for (const node of this.nodes()) {
+      yield node.value;
+    }
+  }
+
+  private *nodes() {
+    let ptr = this.head.next;
+    while (ptr !== this.head) {
+      // Save next before yielding so that we make removing within iteration safe
+      const { next } = ptr;
+      yield ptr;
+      ptr = next;
+    }
+  }
+
+  /** Insert at end of list */
+  push(value: T) {
+    this.count += 1;
+    const newNode: ListNode<T> = {
+      next: this.head as unknown as ListNode<T>,
+      prev: this.head.prev,
+      value
+    };
+    this.head.prev.next = newNode;
+    this.head.prev = newNode;
+  }
+
+  /** Inserts every item inside an iterable instead of the iterable itself */
+  pushMany(iterable: Iterable<T>) {
+    for (const value of iterable) {
+      this.push(value);
+    }
+  }
+
+  /** Insert at front of list */
+  unshift(value: T) {
+    this.count += 1;
+    const newNode: ListNode<T> = {
+      next: this.head.next,
+      prev: this.head as unknown as ListNode<T>,
+      value
+    };
+    this.head.next.prev = newNode;
+    this.head.next = newNode;
+  }
+
+  private remove(node?: ListNode<T> | null) {
+    if (node == null || this.length === 0) {
+      return { next: null, prev: null, value: null };
+    }
+
+    this.count -= 1;
+
+    const prevNode = node.prev;
+    const nextNode = node.next;
+    prevNode.next = nextNode;
+    nextNode.prev = prevNode;
+
+    // Null out the pointers so that a double remove does not break things
+    node.next = null as any;
+    node.prev = null as any;
+    return node;
+  }
+
+  /** Removes the first node at the front of the list */
+  shift() {
+    return this.remove(this.head.next).value;
+  }
+
+  /** Removes the last node at the end of the list */
+  pop() {
+    return this.remove(this.head.prev).value;
+  }
+
+  /** Iterates through the list and removes nodes where filter returns true */
+  prune(filter: (value: T) => boolean) {
+    for (const node of this.nodes()) {
+      if (filter(node.value)) {
+        this.remove(node);
+      }
+    }
+  }
+
+  clear() {
+    this.count = 0;
+    this.head.next = this.head;
+    this.head.prev = this.head;
+  }
+
+  first(): T | null {
+    // If the list is empty, value will be the sentinel's null
+    return this.head.next.value;
+  }
+
+  last(): T | null {
+    // If the list is empty, value will be the sentinel's null
+    return this.head.prev.value;
+  }
+}
 
 /**
  * A pool of Buffers which allow you to read them as if they were one
  * @internal
  */
 export class BufferPool {
-  [kBuffers]: Buffer[];
-  [kLength]: number;
+  private buffers: List<Buffer>;
+  private totalByteLength: number;
```

Comment:
thanks! just for anyone else wondering about this, since this is internal to our driver there isn't a reason to keep properties any more private than TS enforced private, and symbols add a slim amount of overhead that isn't necessary for protecting this class from outside edits at runtime.